portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -54,11 +53,42 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       Comment comment = new Comment(id, name, text, timestamp);
       comments.add(comment);
     }
+    return comments;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    JsonObject jsonResponse = new JsonObject();
+
+    try {
+      int maxComments = Integer.parseInt(request.getParameter("max-comments"));
+      List<Comment> comments = queryComments(maxComments);
+
+      Gson gson = new Gson();
+
+      JsonParser parser = new JsonParser();
+      JsonElement commentsJson = parser.parse(gson.toJson(comments));
+
+      jsonResponse.add("comments", commentsJson);
+      jsonResponse.addProperty("success", true);
+      } catch (NumberFormatException nfe) {
+        jsonResponse.addProperty("success", false);
+        String errorMessage = nfe.getMessage();
 
-    Gson gson = new Gson();
+        /* When NumberFormatException caught because
+        *  max-comments parameter is null */
+        if (errorMessage == "null") {
```

Comment:
I don't think this should be as part of the exception handling. You can check this before hand and return a valid error.